pkg/controller/jobframework/interface.go

```diff
@@ -107,6 +108,15 @@ type JobWithPriorityClass interface {
 	PriorityClass() string
 }
 
+// JobWithValidation optional interface that allows custom webhook validation
+// for Jobs that use BaseWebhook.
+type JobWithValidation interface {
```

Comment:
```suggestion
type JobWithCustomValidation interface {
```

@mbobrovskyi Couldn't we use this name to align with the `JobWithCustomStop`. Indeed, this interface indicate the adding new validations instead of overriding.

Reply:
Done on https://github.com/kubernetes-sigs/kueue/pull/3460.